wordle_solver/wordle_solver.py

```diff
@@ -0,0 +1,66 @@
+from enum import Enum
+from string import ascii_lowercase
+from itertools import groupby
+
+from .word_initializers.english_words_initilalizer import EnglishWords
+
+WORDS_LENGTH = 5
+
+
+class Languages(Enum):
+    ENGLISH = EnglishWords
+
+
+class WordleSolver:
+    def __init__(self, words_length: int, language: str):
+        self.words_length: int = words_length
+        self.all_words: set[str] = Languages[language].value(self.words_length).get_words()
+        self.keys: dict[str, set[int]] = dict()
+        self.must_haves: set[str] = set()
+        self.reset()
+
+    def reset(self):
+        self.keys = {letter: {i for i in range(self.words_length)} for letter in ascii_lowercase}
+        self.must_haves = set()  # each character points to the number of times it could occur.
+
+    def insert_green_letter(self, letter: str, index: int):
+        self.must_haves.add(letter)
+        self.keys[letter] = {index}
+
+    def insert_yellow_letter(self, letter: str, index: int):
+        self.must_haves.add(letter)
+        self.keys[letter].discard(index)
+
+    def insert_black_letter(self, letter: str, index: int):
+        self.keys[letter] = set()
+
+    def valid_word(self, word: str) -> bool:
+        return all([index in self.keys[letter] for index, letter in enumerate(word)]) and \
+               all([letter in word for letter in self.must_haves])
+
+    def guess_word(self, word: str, colors: str) -> bool:
```

Comment:
The capitalization of <word> and <colors> is unintuitive.
I've changed this in my latest work on the CLI branch, such that <word> is automatically made lowercase and <colors> is automatically made capital.
You may want to do this here too, or maybe I'll do it?

Reply:
I think that it should not be automatic here, since when we will have a real product this module will be internal, and in here we would like to have the exact form of input, instead of modifying it many times inside many different functions (in case we add more functionality in the future)

Reply:
I'm afraid that outer viewers/contributors would have trouble testing the file this way.
Maybe there is a good way to signal the capitalization to them?
Maybe even just comments above the methods?